zerver/lib/message.py

```diff
@@ -329,6 +329,11 @@ def finalize_payload(
         if not skip_copy:
             obj = copy.copy(obj)
 
+        if obj["sender_email_address_visibility"] != UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+            # If email address of the sender is only available to administrators,
+            # clients cannot compute gravatars, so we force-set it to false.
+            client_gravatar = False
```

Comment:
We aren't really in a position to do plumbing for this, but I think the comment should perhaps have an added sentence "If we plumbed the current user's role, we could allow client_gravatar=True here if the current user's role has access to the target user's email address"